docker/docker-compose.yml

```diff
@@ -0,0 +1,132 @@
+version: "3.7"
+
+x-logging: &default-logging
+  driver: "json-file"
+  options:
+    max-size: "50m"
+    max-file: "4"
+
+services:
+  proxy:
+    image: nginx:1-alpine
+    ports:
+      # actual and only port facing any connection from outside
+      # Note, change the left number if port 1236 is already in use on your system
+      # You could use port 80 if you won't use a reverse proxy
+      - "1236:1236"
+      - "8536:8536"
+      - "80:80"
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro,Z
+    restart: unless-stopped
+    depends_on:
+      - pictrs
+      - lemmy-ui
+    logging: *default-logging
+
+  lemmy:
+    # use "image" to pull down an already compiled lemmy. make sure to comment out "build".
+    image: dessalines/lemmy:0.19.3
+    # platform: linux/x86_64 # no arm64 support. uncomment platform if using m1.
+    # use "build" to build your local lemmy server image for development. make sure to comment out "image".
+    # run: docker compose up --build
+
+    #build:
+    # context: ../
+    # dockerfile: ./Dockerfile
+    # args:
+    #   RUST_RELEASE_MODE: release
+    #   CARGO_BUILD_FEATURES: default
+    # this hostname is used in nginx reverse proxy and also for lemmy ui to connect to the backend, do not change
+    hostname: lemmy
+    restart: unless-stopped
+    environment:
+      - RUST_LOG="warn,lemmy_server=debug,lemmy_api=debug,lemmy_api_common=debug,lemmy_api_crud=debug,lemmy_apub=debug,lemmy_db_schema=debug,lemmy_db_views=debug,lemmy_db_views_actor=debug,lemmy_db_views_moderator=debug,lemmy_routes=debug,lemmy_utils=debug,lemmy_websocket=debug"
+      - RUST_BACKTRACE=full
+    ports:
+      # prometheus metrics can be enabled with the `prometheus` config option. they are available on
+      # port 10002, path /metrics by default
+      - "10002:10002"
+    volumes:
+      - ./lemmy.hjson:/config/config.hjson:Z
+    depends_on:
+      - postgres
+      - pictrs
+    logging: *default-logging
+
+  lemmy-ui:
```

Comment:
This shouldn't be necessary here.

It should have lemmy-ui-leptos in its place.

Reply:
Or if its best to develop outside of docker (which I'm sure it is), then just leave out this block, and maybe have a comment. And it'd be good to rename this docker-compose-develop.yml or something to show that its only for development.

Reply:
i left the default file name as it makes it easier to run with docker compose, and it is pretty obvious what it's for in this context